### PR TITLE
Update help channel system to use forum channels

### PR DESCRIPTION
**Pull request by ChrisLovering**
This leverages Discord's new forum channel feature, which removes the need for a lot of our custom logic, simplifying the help channel cog significantly.

Sadly this is a rewrite, so the diff isn't very useful. I'd recommend reviewing just the new code.

Marked as do not merge as there are fixup commits I'd like to squash before this is merged.

### CODE REVIEW

**Comment by ChrisLovering** [comment]
Draft until forum channels are fully available.

**Comment by onerandomusername** [comment]
> Draft until forum channels are fully available.

forum channels are out to 90% of community guilds and pydis has early access, why are you waiting for something that has no eta?

**Comment by onerandomusername** [comment]
An important thing to note is that in general, guilds without the community feature don't have the option to have forum channels. So if you're looking on a guild that doesn't have community, it won't have forum channels.

**Comment by ChrisLovering** [comment]
> > Draft until forum channels are fully available.
> 
> forum channels are out to 90% of community guilds and pydis has early access, why are you waiting for something that has no eta?

Because our test server, which is marked as community is yet to have them.

**Comment by onerandomusername** [comment]
Ah, so your test server is in the 10%. Gotcha.

**Review by shtlrs — COMMENTED** [review]
That's a substantial reduction of the amount of code we had.
I unfortunately cannot test this since I don't have access to the forum feature.

I left some comments here & there, thanks Chris !

File: bot/exts/help_channels/_cog.py
Comment (HassanAbouelela) [reply bot/exts/help_channels/_cog.py:88]: This is already possible with the native discord UI, the command is not necessary for the creator.

File: bot/exts/help_channels/_stats.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_stats.py]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_stats.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_stats.py]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_channel.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_channel.py]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_channel.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_channel.py]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_channel.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_channel.py]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_cog.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_cog.py:101]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_cog.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_cog.py:115]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/a3f5978ab40057693aeebfe1bcd57da73462ba6c

File: bot/exts/help_channels/_channel.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_channel.py]: https://github.com/python-discord/bot/pull/2318/commits/74e4d4a1e9b419a210296721c9c51a87f26314cf

File: bot/exts/help_channels/_cog.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_cog.py]: Fixed in https://github.com/python-discord/bot/pull/2318/commits/c3455e6d303ff5b8cd6a7bb2bbdfeef1341f575c

File: bot/exts/help_channels/_cog.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_cog.py:88]: Yea, this is the reason.

**Comment by ChrisLovering** [comment]
Marked as do not merge as there are fixup commits I'd like to squash before this is merged.

File: bot/exts/help_channels/_cog.py
Comment (HassanAbouelela) [inline bot/exts/help_channels/_cog.py:35]: Should we unify the top key for stats? Some are under `help_forum` and some under `help`.

File: bot/exts/help_channels/_cog.py
Comment (HassanAbouelela) [inline bot/exts/help_channels/_cog.py:120]: This get called for every message in every channel, which would cause the owner_id to raise an attribute error. We should remove the if statement below and just return early on `is_help_forum_post` instead.

File: bot/exts/help_channels/_message.py
Comment (HassanAbouelela) [inline bot/exts/help_channels/_message.py:15]: Remove these staticmethod decorators.

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]
Looks and works really nicely :D

Just a few issues

File: bot/exts/help_channels/_cog.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_cog.py:35]: The only reason I kept this one the same is to keep the previous stats together with the new stats

Comment (HassanAbouelela) [reply bot/exts/help_channels/_cog.py:35]: Fair enough, that works for me.

File: bot/exts/help_channels/_channel.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_channel.py]: https://github.com/python-discord/bot/pull/2318/commits/6b4fd8c96d96dad03d5ea294657c1b7ca4aacb11

File: bot/exts/help_channels/_channel.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_channel.py]: https://github.com/python-discord/bot/pull/2318/commits/6b4fd8c96d96dad03d5ea294657c1b7ca4aacb11

File: bot/exts/help_channels/_cog.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_cog.py:120]: https://github.com/python-discord/bot/pull/2318/commits/13ec29de71871cf9e4ae84aeb604931bffecac6d

File: bot/exts/help_channels/_message.py
Comment (ChrisLovering) [reply bot/exts/help_channels/_message.py]: https://github.com/python-discord/bot/pull/2318/commits/84b1de1c1ec5911d91fe5a9f46f2dd9326d4b0eb

**Review by HassanAbouelela — APPROVED** [review]

**Review by shtlrs — APPROVED** [review]
Beautiful work. 
Unfortunately, I didn't test this since I don't have access to the forum feature.

Thanks Chris !

**Review by mbaruh — COMMENTED** [review]
Consider changing CODEOWNERS

**Comment by ChrisLovering** [comment]
Example alert from https://github.com/python-discord/bot/pull/2318/commits/cfec9d96156052e99a3b8f0bf6abb8c01ebfd0dc
image

**Pull request merged**
